addons/website_sale/tests/test_website_sale_cart.py

```diff
@@ -378,3 +378,20 @@ def test_keep_note_line(self):
             self.assertRecordValues(order.order_line, [{
                 "display_type": "line_note",
             }])
+
+    def test_checkout_no_delivery_available(self):
```

Comment:
Unpublish all delivery methods manually so this test doesn't depend on the common test data.

Reply:
Alternatively, mock `_get_preferred_delivery_method` to return no delivery method.